exporter/metric/metric_test.go

```diff
@@ -39,14 +40,20 @@ var (
 )
 
 func TestDescToMetricType(t *testing.T) {
+	cloudMock := cloudmock.NewCloudMock()
+	client := cloudMock.MetricServiceClient
```

Comment:
This is different from the approach in #66 where `[]option.ClientOption{option.WithGRPCConn(mock.ClientConn())}` is used instead.
Shall we use the same approach in both PRs for consistency (whichever is preferrable)?

Reply:
I'm not too sure about the usability of the approach in #66 here since the metric exporters here are defined literally by their respective structs and not by using the `newMetricExporter` function defined [here](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/15b5fb3c8d62367a9f9333ee86b5fbfe0c7e7670/exporter/metric/metric.go#L78). 

Seems to me there is a way to proceed, one involving reverting the changes to `newMetricExporter` and passing in the correct clientOptions to direct requests to our mock api library. After looking closer, it seems newMetricExporter is the more "commerical" method of creating an exporter where one would actually want to connect to the google cloud instance instead of our mock api. Correct me if I'm misunderstanding, but I shouldn't have touched that file entirely.

Reply:
Mocking should be still as close to the actual code being used as possible, so it's preferable to use constructor functions over directly creating structs when possible.

If I understand correctly, using `newMetricExporter` will increase the code coverage in tests, and will test the actual code executed when the metric exporter is being initialized in https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/8fcf64591886d64b6c07975ceed0828fb63a8920/exporter/metric/cloudmonitoring.go#L73

It's fine to merge the change as is, and then you may consider creating another PR switching to `newMetricExporter`.

Reply:
I agree, thanks a lot.